orangecontrib/text/widgets/utils/widgets.py

```diff
@@ -535,9 +535,10 @@ def __init__(self, widget, model_format, provider_format,
 
     def load_model(self, path_to_file):
         self.model_path = path_to_file
-        self.valueChanged.emit(self.model_path, self.resource_path)
+        if self.model_path:
+            self.valueChanged.emit(self.model_path, self.resource_path)
 
     def load_provider(self, path_to_file):
         self.resource_path = path_to_file
-        self.valueChanged.emit(self.model_path, self.resource_path)
-
+        if path_to_file and self.model_path:
+            self.valueChanged.emit(self.model_path, self.resource_path)
```

Comment:
Agree.
It does work but I'm not sure how to handle all if checks without introducing a new level of hierarchy...